wallet/dapps-communication.md

```diff
@@ -8,4 +8,148 @@ Initial meta issue: <https://github.com/massalabs/massa-standards/issues/13>
 
 ## Specification
 
+Here is pseudo code for the two points of view:
+
+- browser extension developer
+- dapp builder
+
+**As a browser extension developer, I would write...**
+
+browser extension code, [content script](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Content_scripts):
+
+```text
+// the extension register itself
+if (window.massaWalletProvider == undefined) { /* wait */ }
+
+window.bearbyWalletProvider = new EventTarget()
+
+window.massaWalletProvider.addEventListener('loaded', () => {
+  window.massaWalletProvider.dispatchEvent(new CustomEvent('register', {
+    ProviderName: "bearby",
+    eventTarget: "bearbyWalletProvider",
+    extensionID: "" // maybe not needed
```

Comment:
extensionID is intended to be used for communication between content_script and background_script, but we need to check the default behavior to determine if it's really needed.